services/housetables/src/main/java/com/linkedin/openhouse/housetables/repository/impl/jdbc/UserTableHtsJdbcRepository.java

```diff
@@ -42,6 +44,33 @@ Optional<UserTableRow> findByDatabaseIdIgnoreCaseAndTableIdIgnoreCase(
   Iterable<UserTableRow> findAllByDatabaseIdAndTableIdLikeAllIgnoreCase(
       String databaseId, String tableIdPattern);
 
+  @Query(
+      "SELECT DISTINCT databaseId FROM UserTableRow u where "
+          + "(:databaseId IS NULL OR lower(u.databaseId) = lower(:databaseId))")
+  Page<String> findAllDistinctDatabaseIds(String databaseId, Pageable pageable);
+
+  Page<UserTableRow> findAllByDatabaseIdIgnoreCase(String databaseId, Pageable pageable);
+
+  Page<UserTableRow> findAllByDatabaseIdAndTableIdLikeAllIgnoreCase(
+      String databaseId, String tableIdPattern, Pageable pageable);
+
+  @Query(
+      "select DISTINCT u from UserTableRow u where "
+          + "(:databaseId IS NULL OR lower(u.databaseId) = lower(:databaseId)) AND "
+          + "(:tableId IS NULL OR lower(u.tableId) = lower(:tableId)) AND "
+          + "(:tableVersion IS NULL OR u.version = :tableVersion) AND "
+          + "(:metadataLocation IS NULL OR u.metadataLocation = :metadataLocation) AND "
+          + "(:storageType IS NULL OR u.storageType = :storageType) AND "
+          + "(:creationTime IS NULL OR u.creationTime = :creationTime)")
+  Page<UserTableRow> findAllByFilters(
```

Comment:
Mind if you can do the same for non-paged findAllByFilters?

Reply:
Yes existing non-page method is first searching all the tables then applying filter which is not good for performance. Yes updated the same using query.